firmware/console/binary/tunerstudio.cpp

```diff
@@ -686,12 +686,23 @@ int TunerStudio::handleCrcCommand(TsChannelBase* tsChannel, char *data, int inco
 
 	const uint16_t* data16 = reinterpret_cast<uint16_t*>(data);
 
-	uint16_t offset = data16[0];
-	uint16_t count = data16[1];
+	uint16_t offset = 0;
+	uint16_t count = 0;
+
+	if (incomingPacketSize >= 3) {
```

Comment:
You don't need this check, data16[0]/[1] are always guaranteed to be inside the write buffer even if they didn't get written this time.

Reply:
(doing it this way adds extra branches whose result is always ignored)

Reply:
Yes. It is guaranteed by current design. Not by function definition.
It is safe to access this memory. But it is not safe to use values that can be random.
Good compiler will not insert branching here, just a few instructions with conditional execution.